### PR TITLE
Remove unwanted characters from gathered legacy facts

### PR DESCRIPTION
**Pull request by GomathiselviS**
Signed-off-by: GomathiselviS <gomathiselvi@gmail.com>

##### SUMMARY
Fixes https://github.com/ansible-collections/vyos.vyos/issues/65

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
common/facts/facts.py

### CODE REVIEW

**Comment by pabelanger** [comment]
recheck

**Comment by mjrider** [comment]
Ilve tested this agains my reproduction for https://github.com/ansible-collections/vyos.vyos/issues/65 and it seems to fix the trailing characters in the configuration except the rest of my facts are now wrong

before patch:
```yaml
    ansible_net_hostname: "router\e[m"
    ansible_net_neighbors: {}
    ansible_net_python_version: 2.7.16
    ansible_net_serialnum: null
    ansible_net_system: vyos
    ansible_net_version: "VyOS 1.3-rolling-202007130117\e[m"
```

after patch:
```yaml
    ansible_net_hostname: null
    ansible_net_neighbors: null
    ansible_net_python_version: 2.7.16
    ansible_net_serialnum: null
    ansible_net_system: null
    ansible_net_version: null
```

also, `ansible_net_commits` is now an array of empty strings instead of an array og hashes

File: plugins/module_utils/network/common/utils.py [hunk 1]
```diff
@@ -771,3 +772,18 @@ def convert_doc_to_ansible_module_kwargs(doc):
...
+        value = "".join(filter(lambda x: x in printable, value))
+        value = re.sub(r"\[m", "", value)
+        return value
+    elif isinstance(value, list):
+        val = []
+        for el in value:
+            v = "".join(filter(lambda x: x in printable, el))
+            v = re.sub(r"\[m", "", v)
+            val.append(v)
+        return val
```
Comment (mjrider) [inline plugins/module_utils/network/common/utils.py:789]: Doesn't this miss a default case where it just returns value ?
For when it's not a string or a list

```suggestion
        return val
      else:
        return value        
```

**Comment by GomathiselviS** [comment]
Hi @mjrider , I am not sure why most of your facts are null. This is what my facts look like after adding the fix. I have added your suggested fix as well. Thanks for reviewing. Can you verify once again?

    "ansible_facts": {
        "ansible_net_api": "cliconf",
        "ansible_net_commits": [
            "",
            "",
            ""
        ],
        "ansible_net_config": [
            "set interfaces ethernet eth0 address 'dhcp'\nset interfaces ethernet eth0 hw-id '08:00:27:83:8e:ce'\nset interfaces ethernet eth1 hw-id '08:00:27:d9:f3:07'\nset interfaces ethernet eth2 hw-id '08:00:27:9c:be:b1'\nset interfaces ethernet eth3 hw-id '08:00:27:96:bb:44'\nset interfaces loopback lo\nset service ssh\nset system config-management commit-revisions '100'\nset system console device ttyS0 speed '115200'\nset system host-name 'vyos'\nset system login user vyos authentication encrypted-password '$6$3pnja.93Z2Po3$Wcg/XiJPS5mYAejbQM9bpSb2TWrTiyVqhCjyri2lp5K.npYfUJG0ksROgy3dXAeSS.NC4v5sGyZYxMIYT9uij.'\nset system login user vyos authentication plaintext-password ''\nset system ntp server 0.pool.ntp.org\nset system ntp server 1.pool.ntp.org\nset system ntp server 2.pool.ntp.org\nset system syslog global facility all level 'info'\nset system syslog global facility protocols level 'debug'",
            "0   2020-07-15 18:51:50 by vyos via cli\n1   2020-07-15 18:49:57 by vyos via cli\n2   2020-07-15 18:48:38 by root via vyos-boot-config-loader\n3   2020-07-15 18:48:37 by root via init"
        ],
        "ansible_net_gather_network_resources": [],
        "ansible_net_gather_subset": [
            "config",
            "default",
            "neighbors"
        ],
        "ansible_net_hostname": "vyos",
        "ansible_net_neighbors": {},
        "ansible_net_python_version": "3.6.5",
        "ansible_net_serialnum": null,
        "ansible_net_system": "vyos",
        "ansible_net_version": "VyOS 1.3-rolling-202007010117",
        "ansible_network_resources": {}
    },
    "changed": false,
    "invocation": {
        "module_args": {
            "gather_network_resources": null,
            "gather_subset": [
                "all"
            ],
            "provider": null
        }
    }
}

**Comment by GomathiselviS** [comment]
Fixed by https://github.com/ansible-collections/vyos.vyos/pull/69. So closing this PR.

**Pull request closed without merging**